src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+                spinner.setEditable(true);
+
+                this.controlPane.getChildren().add(spinner);
+                this.valueProperty.bindBidirectional(spinnerValueFactory.valueProperty());
+
+                break;
+            }
+            case SLIDER: {
+                // To use a slider, the socket should be able to hold a number
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType()));
+
+                // A slider needs a min and max, so the domain should hold two values.
+                checkArgument(domain != null && domain.length == 2);
+                double min = ((Number) domain[0]).doubleValue(),
+                        max = ((Number) domain[1]).doubleValue(),
+                        initialValue = ((Number) socket.getValue()).doubleValue();
+
+                Slider slider = new Slider(min, max, initialValue);
+                this.controlPane.getChildren().add(slider);
+                this.valueProperty.bindBidirectional(slider.valueProperty());
+
+                break;
+            }
+            case RANGE: {
+                // A range produces an array of two numbers, so the socket should expect an array that can hold numbers
+                checkArgument(socketHint.getType().isArray());
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType().getComponentType()));
+
+                // Same
+                checkArgument(domain != null && domain.length == 2);
+                double min = ((Number) domain[0]).doubleValue(),
+                        max = ((Number) domain[1]).doubleValue(),
+                        initialValue = ((Number) socket.getValue()).doubleValue();
+
+                RangeSlider rangeSlider = new RangeSlider(min, max, initialValue, initialValue + 10);
+                this.controlPane.getChildren().add(rangeSlider);
+                this.valueProperty.bindBidirectional(rangeSlider.lowValueProperty());
+
+                break;
+            }
+            case SELECT:
+                // TODO: implement a select control once we have some operations that have enum inputs
+                break;
+        }
+
+        this.controlPane.disableProperty().set(this.socket.isConnected());
+    }
+
+    @Subscribe
+    public void onSocketChanged(SocketChangedEvent event) {
+        if (event.getSocket() == this.socket) {
+            // When the socket changes value, update valueProperty.  valueProperty may be bound to a GUI control that
+            // will change appearance when the socket's value changes, such as a slider.
+            this.valueProperty.setValue(this.socket.getValue());
+        }
+    }
+
+    @Subscribe
+    public void onConnectionAdded(ConnectionAddedEvent event) {
+        // Disable the control for the socket if its value is set by a connection anyway
+        if (event.getConnection().getInputSocket() == this.socket)
+            this.controlPane.disableProperty().set(this.socket.isConnected());
```

Comment:
This is really bad practice. Either keep it on one line or wrap it in braces.
https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/